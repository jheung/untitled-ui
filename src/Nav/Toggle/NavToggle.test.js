import React from "react";
import { render, fireEvent, cleanup } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import { NavToggle } from ".";

describe("NavToggle component", () => {
	afterEach(() => cleanup());

	describe("- default (no children)", () => {
		test("renders correctly and updates aria-expanded when clicked on", async () => {
			const { getByTestId, asFragment } = render(<NavToggle />);

			expect(asFragment()).toMatchSnapshot();

			const navToggleButton = getByTestId("nav-toggle--default");
			await fireEvent.click(navToggleButton);

			expect(navToggleButton).toHaveAttribute("aria-expanded", "true");
		});

		test("with defaultStyling = true adds css module class names", async () => {
			const { asFragment } = render(<NavToggle defaultStyling={true} />);

			expect(asFragment()).toMatchSnapshot();
		});

		test("with custom callback for onToggleBefore and onToggleAfter", async () => {
			const mockToggleBeforeFN = jest.fn();
			const mockToggleAfterFN = jest.fn();

			const { getByTestId } = render(
				<NavToggle onToggleBefore={mockToggleBeforeFN} onToggleAfter={mockToggleAfterFN} />
			);
			const navToggleButton = getByTestId("nav-toggle--default");
			await fireEvent.click(navToggleButton);

			// Returns correct state
			expect(mockToggleBeforeFN).toHaveBeenCalledTimes(1);
			expect(mockToggleBeforeFN).toHaveBeenCalledWith(true);
			expect(mockToggleAfterFN).toHaveBeenCalledTimes(1);
			expect(mockToggleAfterFN).toHaveBeenCalledWith(true);

			await fireEvent.click(navToggleButton);

			expect(mockToggleBeforeFN).toHaveBeenCalledTimes(2);
			expect(mockToggleBeforeFN).toHaveBeenCalledWith(false);
			expect(mockToggleAfterFN).toHaveBeenCalledTimes(2);
			expect(mockToggleAfterFN).toHaveBeenCalledWith(false);
		});
	});

	describe("- with children", () => {
		test("renders correctly with one child element", async () => {
			const { asFragment } = render(
				<NavToggle>
					<span>Icon</span>
				</NavToggle>
			);

			expect(asFragment()).toMatchSnapshot();
		});

		test("renders correctly with one child component with isOpened props", async () => {
			const MockComponent = ({ isOpened }) => <div>{isOpened.toString()}</div>;
			const { asFragment, getByTestId } = render(
				<NavToggle>
					<MockComponent />
				</NavToggle>
			);

			expect(asFragment()).toMatchSnapshot();

			const navToggleButton = getByTestId("nav-toggle");
			await fireEvent.click(navToggleButton);

			expect(navToggleButton).toHaveTextContent(true);
		});

		test("renders correctly with multiple children elements", async () => {
			const { asFragment } = render(
				<NavToggle>
					<span>Icon</span>
					<span>Icon</span>
					<span>Icon</span>
				</NavToggle>
			);

			expect(asFragment()).toMatchSnapshot();
		});
	});
});
