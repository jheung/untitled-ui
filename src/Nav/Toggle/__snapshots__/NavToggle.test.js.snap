// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavToggle component - default (no children) renders correctly and updates aria-expanded when clicked on 1`] = `
<DocumentFragment>
  <button
    aria-controls="navMenu"
    aria-expanded="false"
    aria-label="Toggle navigation"
    class="nav__toggle"
    data-testid="nav-toggle--default"
    type="button"
  >
    <span
      class="nav__toggle-icon nav__toggle-icon--top"
    />
    <span
      class="nav__toggle-icon nav__toggle-icon--middle"
    />
    <span
      class="nav__toggle-icon nav__toggle-icon--bottom"
    />
  </button>
</DocumentFragment>
`;

exports[`NavToggle component - default (no children) with defaultStyling = true adds css module class names 1`] = `
<DocumentFragment>
  <button
    aria-controls="navMenu"
    aria-expanded="false"
    aria-label="Toggle navigation"
    class="nav__toggle toggle"
    data-testid="nav-toggle--default"
    type="button"
  >
    <span
      class="nav__toggle-icon nav__toggle-icon--top toggleIcon toggleIconTop"
    />
    <span
      class="nav__toggle-icon nav__toggle-icon--middle toggleIcon toggleIconMiddle"
    />
    <span
      class="nav__toggle-icon nav__toggle-icon--bottom toggleIcon toggleIconBottom"
    />
  </button>
</DocumentFragment>
`;

exports[`NavToggle component - with children renders correctly with multiple children elements 1`] = `
<DocumentFragment>
  <button
    aria-controls="navMenu"
    aria-expanded="false"
    aria-label="Toggle navigation"
    class="nav__toggle"
    data-testid="nav-toggle"
    type="button"
  >
    <span>
      Icon
    </span>
    <span>
      Icon
    </span>
    <span>
      Icon
    </span>
  </button>
</DocumentFragment>
`;

exports[`NavToggle component - with children renders correctly with one child component with isOpened props 1`] = `
<DocumentFragment>
  <button
    aria-controls="navMenu"
    aria-expanded="false"
    aria-label="Toggle navigation"
    class="nav__toggle"
    data-testid="nav-toggle"
    type="button"
  >
    <div>
      false
    </div>
  </button>
</DocumentFragment>
`;

exports[`NavToggle component - with children renders correctly with one child element 1`] = `
<DocumentFragment>
  <button
    aria-controls="navMenu"
    aria-expanded="false"
    aria-label="Toggle navigation"
    class="nav__toggle"
    data-testid="nav-toggle"
    type="button"
  >
    <span>
      Icon
    </span>
  </button>
</DocumentFragment>
`;
